cultivation/scripts/running/walk_utils.py

```
@@ -0,0 +1,142 @@
+        'valid_segments': valid_segments
+    }
+
+
+def walk_block_segments(gpx_df, is_walk_col, pace_col, cad_col, cad_thr=128, max_gap_s=2, min_dur_s=2):
+    """
+    Group contiguous walk blocks after filtering GPS jitter, allowing up to `max_gap_s` seconds of non-walk between walk intervals.
+    Returns a list of segment dicts.
+    """
+    walk_df = gpx_df[gpx_df[is_walk_col]].copy()
+    walk_df = filter_gps_jitter(walk_df, pace_col, cad_col, cad_thr)
+    is_walk = gpx_df[is_walk_col].astype(int)
```

Ruff (0.8.2)

Local variable `is_walk` is assigned to but never used

Remove assignment to unused variable `is_walk`

(F841)